pandas/core/base.py

```diff
@@ -1201,7 +1205,13 @@ def _map_values(self, mapper, na_action=None):
             else:
                 values = self.values
 
-            indexer = mapper.index.get_indexer(values)
+            try:
+                indexer = mapper.index.get_indexer(values)
+            except InvalidIndexError:
```

Comment:
So this is just catching all Exceptions right? Can we be more explicit?

Reply:
@WillAyd : This is catching a specific subclass of `Exception`.  I'm not sure how this would be a catch-all.  This looks fine to me here.